pyrdp/mitm/FastPathMITM.py

```diff
@@ -13,14 +14,15 @@ class FastPathMITM:
     MITM component for the fast-path layer.
     """
 
-    def __init__(self, client: FastPathLayer, server: FastPathLayer):
+    def __init__(self, client: FastPathLayer, server: FastPathLayer, state: RDPMITMState):
         """
         :param client: fast-path layer for the client side
         :param server: fast-path layer for the server side
         """
 
         self.client = client
         self.server = server
+        self.state = state
```

Comment:
I'm pretty sure that's the most complex file we have, because of the protocol. It's still only about 500 lines though, which isn't that bad IMO. Once you understand the basics of the DeviceRedirectionMITM (first ~300 lines), I think understanding what the rest is about isn't that hard. It's just classes for handling the forged directory listing / file read requests.

Reply:
Sorry, wrong comment. For why it's used, check the other methods :P